src/server/appserver.cpp

```diff
@@ -2003,6 +2060,92 @@ void AppServer::sendErrorsCleared(int syncDbId) {
     CommServer::instance()->sendSignal(SIGNAL_NUM_UTILITY_ERRORS_CLEARED, params, id);
 }
 
+void AppServer::sendLogUploadStatusUpdated(char status, int percent) {
+    int id;
+
+    QByteArray params;
+    QDataStream paramsStream(&params, QIODevice::WriteOnly);
+    paramsStream << status;
+    paramsStream << percent;
+    CommServer::instance()->sendSignal(SIGNAL_NUM_UTILITY_LOG_UPLOAD_STATUS_UPDATED, params, id);
+
+    std::string logUploadStatus;  // Save progress in the database
+    logUploadStatus += status + std::to_string(percent);
+    if (bool found = false;
+        !ParmsDb::instance()->updateAppState(AppStateKey::LogUploadStatus, logUploadStatus, found) || !found) {
+        LOG_WARN(_logger, "Error in ParmsDb::updateAppState with key=" << static_cast<int>(AppStateKey::LogUploadStatus));
+        // Don't fail because it is not a critical error, especially in this context where we are trying to send logs
+    }
+}
+
+void AppServer::cancelLogUpload() {
+    ExitCause exitCause = ExitCauseUnknown;
+    ExitCode exitCode = ServerRequests::cancelLogToSupport(exitCause);
+    if (exitCode == ExitCodeOperationCanceled) {
+        LOG_WARN(_logger, "Operation already canceled");
+        sendLogUploadStatusUpdated('C', 1);
```

Comment:
Again no hardcoded string

Reply:
Will be change with this comment: https://github.com/Infomaniak/desktop-kDrive/pull/117#discussion_r1601570338